### PR TITLE
Update tables for wig180 corrs

### PR DESCRIPTION
**Pull request by xresende**
- Updated pwrsupply tables according to @gabrielbrunheira suggestions on adding wiggler180 correctors.
- We still need to update the excitation curve for these correctors! (@Gabrielrezende-asc ) (**DONE!**)

### CODE REVIEW

File: beaglebone/udc-bsmp.txt [hunk 1]
```diff
@@ -226,7 +227,7 @@ IA-10RaPS01:PS-UDC-SI3      SI-10M1:PS-QS 7     SI-10M2:PS-QS 8
 IA-11RaPS01:PS-UDC-SI3      SI-11M1:PS-QS 7     SI-11M2:PS-QS 8
 IA-12RaPS01:PS-UDC-SI3      SI-12M1:PS-QS 6     SI-12M2:PS-QS 7
 IA-13RaPS01:PS-UDC-SI3      SI-13M1:PS-QS 6     SI-13M2:PS-QS 7
-IA-14RaPS01:PS-UDC-SI3      SI-14M1:PS-QS 7     SI-14M2:PS-QS 8
+IA-14RaPS01:PS-UDC-SI3      SI-14M1:PS-QS 7     SI-14M2:PS-QS 8     SI-14SB:PS-CH-1 9     SI-14SB:PS-CH-2 10
```
[the remark this answers is not in the document]
Comment (gabrielbrunheira) [reply beaglebone/udc-bsmp.txt:230]: @xresende, in discussion with Bruno Limeira, we decided to keep it as following:

IA-14RaPS01:PS-UDC-SI6    SI-14SB:PS-CH-1 9     SI-14SB:PS-CH-2 10

**Review by gabrielbrunheira — CHANGES_REQUESTED** [review]

File: beaglebone/udc-bsmp.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (xresende) [reply beaglebone/udc-bsmp.txt:230]: I see. ok, I will commit an update, @gabrielbrunheira !

**Review by gabrielbrunheira — APPROVED** [review]

File: timesys/high-level-triggers.py [hunk 2]
```diff
@@ -136,6 +136,8 @@
             'SI-13C3:PS-CV-2:BCKPLN',
             'SI-13C4:PS-CH:BCKPLN', 'SI-13C4:PS-CV:BCKPLN',
 
+            'SI-14SB:PS-CH-1:BCKPLN', 'SI-14SB:PS-CV-1:BCKPLN',
+            'SI-14SB:PS-CH-2:BCKPLN', 'SI-14SB:PS-CV-2:BCKPLN',
```
Comment (gabrielbrunheira) [inline timesys/high-level-triggers.py:140]: @fernandohds564 what are these vertical correctors?

**Review by gabrielbrunheira — CHANGES_REQUESTED** [review]

File: timesys/high-level-triggers.py (hunk 2 above)
Comment (gabrielbrunheira) [reply timesys/high-level-triggers.py:140]: I believe they don't exist, only horizontal correctors.

Comment (xresende) [reply timesys/high-level-triggers.py:140]: @fernandohds564 , wiggler two correctors are only for the horizontal plane.

**Comment by xresende** [comment]
@fernandohds564 , I think it is useful to put the URL for `Cabos_e_Fibras_Sirius.xlsx` in the header of the file. This way we can access the original table when in CNPEM network.

File: timesys/high-level-triggers.py (hunk 2 above)
Comment (fernandohds564) [reply timesys/high-level-triggers.py]: ok, I removed them.

**Review by gabrielbrunheira — APPROVED** [review]

**Review by RafaelLyra8 — APPROVED** [review]

**Comment by fernandohds564** [comment]
> @fernandohds564 , I think it is useful to put the URL for `Cabos_e_Fibras_Sirius.xlsx` in the header of the file. This way we can access the original table when in CNPEM network.

@xresende, this is a google docs file. If I put its URL in the header this file everybody with access to github would be able to view and edit it, right?

**Comment by anacso17** [comment]
> > @fernandohds564 , I think it is useful to put the URL for `Cabos_e_Fibras_Sirius.xlsx` in the header of the file. This way we can access the original table when in CNPEM network.
> 
> @xresende, this is a google docs file. If I put its URL in the header this file everybody with access to github would be able to view and edit it, right?

Is there a way to share the file with read-only option?

**Review by anacso17 — APPROVED** [review]

**Comment by fernandohds564** [comment]
right @anacso17. I pushed one commit with this change, @xresende.

**Review by fernandohds564 — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Review by Gabrielrezende-asc — APPROVED** [review]
Excitation data added!

**Comment by xresende** [comment]
@fernandohds564 , did you manage to add it with read-only attribute URL ?

Since the files belongs to the Sirius control system / subsystems, is it not a good idea to ask the maintainer to move its primary location to one of our standard systems ? github, gitlab or sharepoint? are there other relevant Sirius files already stored at google services?

File: magnet/excitation-data/si-corrector-wig180.txt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+# units             Ampere  T*m T*m
+
+# EXCITATION DATA
+# ===============
+-005.000  -1.8111e-03 +0.0000e+00
++005.000  +1.8522e-03 +0.0000e+00
+
+# COMMENTS
+# ========
+# 1. tables needs to be updated with measurement!
```
Comment (anacso17) [inline magnet/excitation-data/si-corrector-wig180.txt:16]: need update?

Comment (xresende) [reply magnet/excitation-data/si-corrector-wig180.txt:16]: an update (also deleting this comment) is coming up...

**Comment by fernandohds564** [comment]
> @fernandohds564 , did you manage to add it with read-only attribute URL ?

yes

> Since this file belongs to the Sirius control system / subsystems, is it not a good idea to ask the maintainer to move its primary location to one of our standard systems ? github, gitlab or sharepoint? are there other relevant Sirius files already stored at google services?

I already asked this in the past, but we could try again. What do you guys think @danielot, @mmdonatti ? 

@xresende, is there any api to communicate with excel files in sharepoint the way I do in siriuspy with google docs?

**Review by anacso17 — CHANGES_REQUESTED** [review]
Hi guys, I think it is missing to add SI-14SB:MA-CH-1 and SI-14SB:MA-CH-2 to magnets-model-data.txt file

**Comment by xresende** [comment]
> Hi guys, I think it is missing to add SI-14SB:MA-CH-1 and SI-14SB:MA-CH-2 to magnets-model-data.txt file

done, @anacso17 !

**Comment by xresende** [comment]
> > @fernandohds564 , did you manage to add it with read-only attribute URL ?
> 
> yes
> 
> > Since this file belongs to the Sirius control system / subsystems, is it not a good idea to ask the maintainer to move its primary location to one of our standard systems ? github, gitlab or sharepoint? are there other relevant Sirius files already stored at google services?
> 
> I already asked this in the past, but we could try again. What do you guys think @danielot, @mmdonatti ?
> 
> @xresende, is there any api to communicate with excel files in sharepoint the way I do in siriuspy with google docs?

there is a python api to interact with sharepoint objects, like retrieving a list of files names and files @ronecastro has implemented a subset. see documentation/procedures/build-link-page.py in PR https://github.com/lnls-sirius/control-system-constants/pull/256

as for getting into spreadsheet elements, I think the file can be retrieves from using api sharepy and spreadsheet can be read/modifies with panda: https://pythonbasics.org/read-excel/

**Comment by fernandohds564** [comment]
I talked to @danielot this morning about this.

He said the reason this table was in google docs is because it used to depend on references for spreadsheets in different files, which was not supported by Sharepoint back then. However, it seems these dependencies doesn't exist anymore, and we could move them. 

He went even further, saying that we could translate this spreadsheet to other formats, such as text files, so it would be in accordance with our standards. 
He also suggested changing all the text files to json or some other multi-plataform format. The advantage would be that we wouldn't need to implement parsing scripts for every file.

Is this a good summary of our talk @danielot ?

I see the advantage of using json, but I wonder if this change would exchange human readability by computer readability and what would be better: to use a format easy to read or easy to maintain...

**Comment by danielot** [comment]
> @xresende, I talked to @danielot this morning about this.
> 
> He said the reason this table was in google docs is because it used to depend on references for spreadsheets in different files, which was not supported by Sharepoint back then. However, it seems these dependencies doesn't exist anymore, and we could move them.
> 
> He went even further, saying that we could translate this spreadsheet to other formats, such as text files, so it would be in accordance with our standards. He also suggested changing all the text files in this repository to json or some other multi-plataform format. The advantage would be that we wouldn't need to implement parsing scripts for every file.
> 
> Is this a good summary of our talk @danielot ?
> 
> I see the advantage of using json, but I wonder if this change would exchange human readability by computer readability and what would be better: to use a format easy to read or easy to maintain...

@xresende and @fernandohds564, I think this is a good summary. I just have one complement in the topic about JSON and human readability. YAML format may be the one to give you what you're looking for. Besides human readability it is more flexible to things like commenting lines and so on.

My main point is that by using text files we are recreating a database. There should be benefits of using standard databases to store such data. I think it should be possible to keep it simple if we adopt a nosql database approach...

Does @edupcoelho have an opinion about this?

**Comment by xresende** [comment]
> > @xresende, I talked to @danielot this morning about this.
> > He said the reason this table was in google docs is because it used to depend on references for spreadsheets in different files, which was not supported by Sharepoint back then. However, it seems these dependencies doesn't exist anymore, and we could move them.
> > He went even further, saying that we could translate this spreadsheet to other formats, such as text files, so it would be in accordance with our standards. He also suggested changing all the text files in this repository to json or some other multi-plataform format. The advantage would be that we wouldn't need to implement parsing scripts for every file.
> > Is this a good summary of our talk @danielot ?
> > I see the advantage of using json, but I wonder if this change would exchange human readability by computer readability and what would be better: to use a format easy to read or easy to maintain...
> 
> @xresende and @fernandohds564, I think this is a good summary. I just have one complement in the topic about JSON and human readability. YAML format may be the one to give you what you're looking for. Besides human readability it is more flexible to things like commenting lines and so on.
> 
> My main point is that by using text files we are recreating a database. There should be benefits of using standard databases to store such data. I think it should be possible to keep it simple if we adopt a nosql database approach...
> 
> Does @edupcoelho have an opinion about this?

thanks, @fernandohds564 and @danielot for the discussion. I think we could use YAML. maybe new tables may be nserted already in this format and older ones gradually migrated. for a quick reference on yaml/pyhton for those of us who never used it: https://zetcode.com/python/yaml/

**Review by anacso17 — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

**Pull request merged**